resolwe_bio/tools/basespace_download.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python3
+"""Tool to download files from BaseSpace."""
+
+import sys
+import atexit
+import argparse
+import requests
+
+
+class BaseSpaceDownloadError(Exception):
+    """BaseSpace download error."""
+    
+    pass
+
+
+def main():
+    """Entry point."""
+    session = requests.Session()
+    atexit.register(on_exit, session)
+
+    parser = argparse.ArgumentParser(description='Download file from Illumina BaseSpace.')
+    parser.add_argument('--file-id',
+                        dest='file_ids',
+                        action='append',
+                        required=True,
+                        help='BaseSpace file ID. This argument can be repeated to specify multiple files.')
+    parser.add_argument('--access-token-secret-path',
+                        dest='access_token_secret_path',
+                        required=True,
+                        help='BaseSpace access token secret path.')
+    args = parser.parse_args()
+
+    file_ids = args.file_ids
+    access_token = get_token_from_secret_file(args.access_token_secret_path)
+    headers = {'x-access-token': access_token}
+
+    for file_id in file_ids:
+        file_name = get_file_name(session, file_id, headers)
+        download_file(session, file_id, file_name, headers)
+        print('filename={}'.format(file_name))
+
+
+def on_exit(session):
+    """Clean up function called on exit."""
+    session.close()
```

Comment:
Why is this needed?

Reply:
Not needed per se, but since `Session` instance offers this method, it might be a good idea to call it.